Feature/occupancygrid filter

TomohitoAndo opened the pull request:
## Status
**PRODUCTION**

## Description
Add following nodes
- grid_map_filter
filter OccupancyGrid with some functions. I will use this for planning.
- wayarea2grid
convert wayare in Vector Map to OccupancyGrid

Added new package:
- Road Occupancy Processor

## Todos
- [x] Tests
  - [x] Ubuntu 14.04
  - [x] Ubuntu 16.04
- [x] Documentation


## Steps to Test or Reproduce
launch from Runtime Manager
Please see following videos for the details.

### grid_map_filter
[![](https://img.youtube.com/vi/bl-nG8Zv-C0/0.jpg)](https://www.youtube.com/watch?v=bl-nG8Zv-C0)
### wayarea2grid
[![](https://img.youtube.com/vi/UkcO4V-0NOw/0.jpg)](https://www.youtube.com/watch?v=UkcO4V-0NOw)

TomohitoAndo committed:
Add grid map filter node

TomohitoAndo committed:
Add wayarea2grid node

TomohitoAndo committed:
Replace dist_transform with grid_map_filter

TomohitoAndo committed:
Add Runtime Manager UI for grid_map_filter node

TomohitoAndo committed:
Add UI for wayarea2grid node

TomohitoAndo committed:
Add instruction videos

TomohitoAndo committed:
Fix videos

TomohitoAndo committed:
Both node handles were private

yk-fujii commented on ros/src/computing/perception/semantics/packages/object_map/nodes/grid_map_filter/grid_map_filter.cpp:
```diff
@@ -0,0 +1,281 @@
+    {
+      // actual distance [meter]
+      double dist = dt_image.at<float>(y, x) * resolution;
+      if (dist > max_dist)
```
this should be in {}

yk-fujii reviewed (approved):
@TomohitoAndo 
Good. I have confirmed to run successfully with ubuntu16.04/14.04

but, please put parentheses to if/for.
I will approve so please merge when that fix is done.

amc-nu committed:
Merge branch 'develop' into feature/occupancygrid_filter

amc-nu committed:
Added Comments Documentation
Code refactoring to follow standards
Added libraries
Separation of Vectormap Processing inside Clustering

amc-nu committed:
Added documentation

amc-nu committed:
Changed variable name

amc-nu committed:
Merge branch 'feature/occupancygrid_filter' of https://github.com/CPFL/Autoware into feature/occupancygrid_filter

amc-nu committed:
Added Road Occupancy Processor package

amc-nu committed:
Added extra documentation
Added commands to RunTimeManager

amc-nu reviewed (approved):
Thanks @TomohitoAndo
